modules/internal/ChapelLocale.chpl

```diff
@@ -530,8 +530,21 @@ module ChapelLocale {
   }
 
 
+  pragma "no doc"
+  proc chpl_getSingletonLocaleArray(arg: locale) const ref
+  lifetime return c_sublocid_none // indicate return has global lifetime
+  {
+    var casted = arg._instance:borrowed AbstractLocaleModel?;
+    if casted == nil then
+      halt("cannot call chpl_getSingletonCurrentLocaleArray on nil or rootLocale");
+
+    return casted!.chpl_singletonThisLocaleArray;
+  }
+
   pragma "no doc"
   class AbstractLocaleModel : BaseLocale {
+    var chpl_singletonThisLocaleArray:[0..0] locale;
```

Comment:
Can you add a comment on this field or `chpl_getSingletonLocaleArray` explaining that this is necessary to avoid creating and copying a new array for DR's targetLocales?

Reply:
Done